client/options.go

```diff
@@ -41,16 +41,16 @@ func WithOpenCensusTracing() Option {
 // so that it will automatically inject trace-headers.
 //
 // Should be used when you trace your application with Datadog.
-func WithDatadogTracing(serviceName string) Option {
+func WithDatadogTracing(opts ...dd_http.RoundTripperOption) Option {
```

Comment:
Is serviceName optional now? If not maybe keep the serviceName argument and just add the ability to append other options?

Reply:
There is multiple ways to set a service name, like for instance starting the tracer:
```
ddtracer.Start(ddtrace.WithService("my-service"))
...
```

This solution will make it possible to both set the service name by options in this function, using starting the tracer or other ways

Reply:
in the case of lambda, you can't set it on the tracer, nor the agent, so we need to set it through options, I believe it needs to be clear somewhere that to use this with a lambda you need to add the service-name option

Reply:
@lonnblad Any suggestions how to make this clear?

Reply:
maybe add a line in the description of the function, that when using this in a lambda, make sure to set the options and then an example of how to set it

Reply:
Updated.